src/main/java/cn/nukkit/level/generator/populator/impl/PopulatorOreEmerald.java

```diff
@@ -0,0 +1,35 @@
+package cn.nukkit.level.generator.populator.impl;
+
+import cn.nukkit.block.Block;
+import cn.nukkit.level.ChunkManager;
+import cn.nukkit.level.format.FullChunk;
+import cn.nukkit.level.generator.populator.type.Populator;
+import cn.nukkit.math.NukkitMath;
+import cn.nukkit.math.NukkitRandom;
+
+import static cn.nukkit.block.BlockID.STONE;
+
+/**
+ * @author GoodLucky777
+ */
+public class PopulatorOreEmerald extends Populator {
+
+    @Override
+    public void populate(ChunkManager level, int chunkX, int chunkZ, NukkitRandom random, FullChunk chunk) {
+        int sourceX = chunkX << 4;
+        int sourceZ = chunkZ << 4;
+        int endX = sourceX + 15;
+        int endZ = sourceZ + 15;
+        
+        for (int i = 0; i < 11; i++) {
+            int x = NukkitMath.randomRange(random, sourceX, endX);
+            int z = NukkitMath.randomRange(random, sourceZ, endZ);
+            int y = NukkitMath.randomRange(random, 4, 31);
```

Comment:
According to Wiki https://minecraft.gamepedia.com/Emerald_Ore#Natural_generation the Y range should be 0 - 32, is that right?

Reply:
it is weird, according to https://minecraft.gamepedia.com/Mineral_vein it was different

Reply:
"This section is missing information about generation of emerald ore (see [here](https://gist.github.com/tryashtar/2a22bd04394275fefb6b49c348758fd6)), does this match up with the values provided on the table below?." If you enter in the url (gist) you see the math of the Y (wich is a random with max 28 + 4 which gives a range of 4 - 32)

Reply:
oh i see, i was confused